src/main/java/org/sagebionetworks/web/client/widget/entity/controller/SynapseAlertImpl.java

```diff
@@ -0,0 +1,137 @@
+import org.sagebionetworks.web.client.utils.Callback;
+import org.sagebionetworks.web.client.widget.entity.JiraURLHelper;
+import org.sagebionetworks.web.shared.exceptions.BadRequestException;
+import org.sagebionetworks.web.shared.exceptions.ForbiddenException;
+import org.sagebionetworks.web.shared.exceptions.NotFoundException;
+import org.sagebionetworks.web.shared.exceptions.ReadOnlyModeException;
+import org.sagebionetworks.web.shared.exceptions.SynapseDownException;
+import org.sagebionetworks.web.shared.exceptions.UnauthorizedException;
+import org.sagebionetworks.web.shared.exceptions.UnknownErrorException;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+public class SynapseAlertImpl implements SynapseAlert, SynapseAlertView.Presenter {
+	GlobalApplicationState globalApplicationState;
+	AuthenticationController authController;
+	SynapseAlertView view;
+	SynapseJSNIUtils synapseJSNIUtils;
+	Throwable ex;
+	
+	@Inject
+	public SynapseAlertImpl(
+			SynapseAlertView view,
+			GlobalApplicationState globalApplicationState,
+			AuthenticationController authController,
+			SynapseJSNIUtils synapseJSNIUtils
+			) {
+		this.view = view;
+		this.globalApplicationState = globalApplicationState;
+		this.authController = authController;
+		this.synapseJSNIUtils = synapseJSNIUtils;
+		view.setPresenter(this);
+		view.clearState();
+	}
+
+	@Override
+	public void handleException(Throwable ex) {
+		clearState();
+		this.ex = ex;
+		synapseJSNIUtils.consoleError(DisplayUtils.getStackTrace(ex));
+		boolean isLoggedIn = authController.isLoggedIn();
+		if(ex instanceof ReadOnlyModeException) {
+			view.showError(DisplayConstants.SYNAPSE_IN_READ_ONLY_MODE);
+		} else if(ex instanceof SynapseDownException) {
+			globalApplicationState.getPlaceChanger().goTo(new Down(DEFAULT_PLACE_TOKEN));
+		} else if(ex instanceof UnauthorizedException) {
+			// send user to login page				
+			view.showInfo(DisplayConstants.SESSION_TIMEOUT, DisplayConstants.SESSION_HAS_TIMED_OUT);
+			globalApplicationState.getPlaceChanger().goTo(new LoginPlace(LoginPlace.LOGOUT_TOKEN));
+		} else if(ex instanceof ForbiddenException) {			
+			if(!isLoggedIn) {
+				view.showLoginAlert();
+			} else {
+				view.showError(DisplayConstants.ERROR_FAILURE_PRIVLEDGES + " " + ex.getMessage());
+			}
+		} else if(ex instanceof BadRequestException) {
+			//show error (not to file a jira though)
+			view.showError(ex.getMessage());
+		} else if(ex instanceof NotFoundException) {
+			view.showError(DisplayConstants.ERROR_NOT_FOUND  + " " + ex.getMessage());
+		} else if (ex instanceof UnknownErrorException) {
+			//An unknown error occurred. 
+			//Exception handling on the backend now throws the reason into the exception message.  Easy!
+			view.showError(ex.getMessage());
+			if (isLoggedIn) {
+				view.showJiraDialog(ex.getMessage());
+			}
+		} else {
+			//not recognized
+			view.showError(ex.getMessage());
```

Comment:
if (ex.getMessage()) is null, empty, or "0" (socket timeout?), show that we did not get a response from the server, could be related to your network connection.